src/glob.rs

```diff
@@ -44,7 +45,7 @@ fn determine_type(glob: &str) -> GlobType {
     }
 }
 
-#[derive(Clone, PartialEq)]
+#[derive(Clone, PartialEq, Eq, Hash)]
```

Comment:
To match the xdgmime implementation, the hashing should be performed on the GlobType and the Mime fields of the Glob structure; the weight should not be taken into consideration, because we might have overrides that increase it.

Reply:
So if you have two identical globs but with different weights, you want to keep the one with the highest weight? I don't think it is possible to ensure that with `HashSet` :confused: 

And what about `PartialEq`? You use the derive implementation which will also use the weight field to compare the globs.

Reply:
Sorry, I wasn't really clear.

> So if you have two identical globs but with different weights, you want to keep the one with the highest weight? 

The existing xdgmime implementation discards any new glob with the same GlobType and MIME type. The weight is not taken into consideration.

If there are two glob definitions with the same MIME type but different GlobType, a new Glob is created; the same happens if there are two glob definitions with the same GlobType but a different MIME type.